src/hydra/dataProvider.js

```diff
@@ -72,7 +74,7 @@ export const transformJsonLdDocumentToReactAdminDocument = (
           false,
         );
       }
-      document[key] = document[key]['@id'];
+      document[key] = useEmbedded ? document[key] : document[key]['@id'];
```

Comment:
```suggestion
      document[key.toString()] = useEmbedded ? document[key.toString()] : document[key.toString()]['@id'];
```
To avoid Object Injection Sink IMO

Reply:
It cannot be done like this because React Admin data providers need to give record objects: https://marmelab.com/react-admin/DataProviders.html#response-format

Reply:
It doesn't affect return, your key object might be a string right ?

Reply:
Oh yes, you're right sorry. But `Object.keys` returns only strings so I don't think it's necessary.